remote-messaging/remote-messaging-impl/src/main/java/com/duckduckgo/remote/messaging/impl/RealRemoteMessageModel.kt

```diff
@@ -68,8 +68,11 @@ class RealRemoteMessageModel @Inject constructor(
         return remoteMessage.content.getSecondaryAction()
     }
 
-    override fun onActionClicked(remoteMessage: RemoteMessage): Action? {
+    override suspend fun onActionClicked(remoteMessage: RemoteMessage): Action? {
         remoteMessagingPixels.fireRemoteMessageActionClickedPixel(remoteMessage)
+        withContext(dispatchers.io()) {
+            remoteMessagingRepository.dismissMessage(remoteMessage.id)
+        }
```

Comment:
@LukasPaczos this is what I meant on Asana. Just add an condition here: if action != SHARE as this would break that functionality